File: py/spoj_iwgbs.py
def comb(n,k):
	i=1
	p=1
	if(k>n-k):
		k=n-k;
			
	for i in range(1,k+1):
		p*=n
		n=n-1
		p/=i
	return p

	
count=1;
n=int(raw_input())
	
for i in range(n/2,n):
	count+=comb(i+1,n-i);
	
print count

File: c,c++/spoj/spoj_iwgbs.c
#include<stdio.h>

long long int comb(int n,int k)
{
	long long int i=1,p=1;
	if(k>n-k)
			k=n-k;
			
		for(i=1;i<=k;++i)
		{
			p*=n--;
			p/=i;
		}
	return p;
}

int main()
{
	int n,i;
	long long int count;
	
	count=1;
	scanf("%d",&n);
	
	for(i=n/2;i<n;++i)
	{
		count+=comb(i+1,n-i);
	}
	
	printf("%lld\n",count);

	return 0;
}
